scRNA-seq/04-scrnsaseq_exercise.Rmd

```diff
@@ -0,0 +1,283 @@
+You can use an `%in%` operator to check if the gene you are looking for is in 
+our filtered gene matrix (Remember it has to be an Ensembl gene ID e.g.`ENSG`,
+and it needs to be in quotes).
+
+```{r eval=FALSE}
+# Use the phrase below to find out if a gene you are interested in is in sce_norm_df
+<ENSEMBL_GENE_ID> %in% sce_norm_df$genes
+```
+
+Use this code chunk to look at what `markers` looks like.
+
+```{r}
+
+```
+
+Let's build a `dplyr` pipe to create a `data.frame` with the data for our marker 
+genes and add on `cell_type` vector as its own column.
+Remember that each step in your pipe needs to be followed by the `%>%` operator
+and that we haven't loaded the `dplyr` library so use the `::` if needed. 
+
+```{r}
+# Make a data.frame with only the genes in `markers`
+marker_data <- sce_norm_df %>%
+  # Use this line to filter the data to only the marker genes we care about.
+  # Use this line for tibble::column_to_rownames function to hide the gene names
+  # Use this line to transpose this data
+  as.data.frame() %>% # We've done this step of turning the matrix back into a 
+                      # data.frame after we transposed
+  # Use this line to add a column with the information in `cell_types`
+```
+
+## Plot the marker genes' expression as a density plot by cell type
+
+Use the following `ggplot` template to make a density plot by cell type for one
+of your genes of interest (hint: use the `marker_data` we just made).
+Remember that for each step in making a `ggplot`, you need to add a `+`.
+
+```{r eval=FALSE}
+# Plot the gene data by cell type using `facet_wrap`
+ggplot2::ggplot(<DATA_FRAME>, ggplot2::aes(x = <GENE_DATA_COLUMN>)) +
+  ggplot2::geom_density(fill = <COLOR_NAME>) + # Choose a color 
+  ggplot2::facet_wrap(~ <CELL_TYPE_INFO>) # Use cell types in this function to 
+                                          # separate the density plots
+```
+
+Feel free to play around with aspects of this plot and see how the plot changes.
+Look at [ggplot2's guide](https://ggplot2.tidyverse.org/) for a list of the 
+options. 
+
+## Make a heatmap of mean expression of marker genes by cell type
+
+Let's build a `dplyr` pipe to calculate the means of these marker genes based on 
+the cell-type labels. 
+
+```{r}
+# Build a pipe to calculate the mean expressions within each cell type
+marker_means <- marker_data %>%
+  # Use this line for a dplyr function to group the data by their cell types 
+  # Use this line for a dplyr function to summarize to a mean
+  # Use the tibble::column_to_rownames hide the cell-type column
+```
+
+Let's change the Ensembl gene IDs to symbols since they are easier to read in 
+the heatmap.
+ 
+```{r eval=FALSE}
+# Change the column names to gene symbols
+colnames(marker_means) <- 
+```
+
+Use `pheatmap::pheatmap` to make a heatmap of cell-type mean expression of these
+marker genes. 
+Use the argument `scale = "row"`. 
+
+```{r eval=FALSE}
+# Make a heatmap of gene expression means and cell types
+
```

Comment:
There is a transpose step in the answer and I would not know why from the text in this notebook.

Reply:
Added an extra instruction for this.